migrator/migrations/m_192_to_m_193_report_config_migrations/migration_impl.go

```diff
@@ -0,0 +1,310 @@
+	rowsNotifier, err := queryv1.Rows()
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to iterate table notifiers")
+	}
+	defer func() { _ = rowsNotifier.Close() }()
+	if rowsNotifier.Next() {
+		return true, nil
+
+	}
+	return false, nil
+}
+
+func getMigratedReportConfigIfExsists(reportID string, db *gorm.DB, dbctx context.Context) (bool, *storage.ReportConfiguration, error) {
+	queryv1 := db.WithContext(dbctx).Table(updatedSchema.ReportConfigurationsTableName).Select("serialized").Where(&updatedSchema.ReportConfigurations{ID: getDeterministicID(reportID)})
+	rowsV1config, err := queryv1.Rows()
+	if err != nil {
+		return false, nil, errors.Wrapf(err, "failed to iterate table %s", "report_configurations")
+	}
+	defer func() { _ = rowsV1config.Close() }()
+	for rowsV1config.Next() {
+		var reportConfigv2 *updatedSchema.ReportConfigurations
+		if err = db.ScanRows(rowsV1config, &reportConfigv2); err != nil {
+			return false, nil, errors.Wrapf(err, "failed to scan rows")
+		}
+		reportv2ConfigProto, err := updatedSchema.ConvertReportConfigurationToProto(reportConfigv2)
+		if err != nil {
+			return false, nil, errors.Wrapf(err, "failed to convert %+v to proto", reportv2ConfigProto)
+		}
+		return true, reportv2ConfigProto, nil
+
+	}
+	return false, nil, nil
+}
+
+func migrate(database *types.Databases) error {
+
+	db := database.GormDB
+	pgutils.CreateTableFromModel(database.DBCtx, db, updatedSchema.CreateTableReportConfigurationsStmt)
+	pgutils.CreateTableFromModel(database.DBCtx, db, updatedSchema.CreateTableReportSnapshotsStmt)
+	pgutils.CreateTableFromModel(database.DBCtx, db, updatedSchema.CreateTableReportConfigurationsStmt.Children[0])
+	db = db.WithContext(database.DBCtx)
+	return db.Transaction(func(tx *gorm.DB) error {
+		query := tx.WithContext(database.DBCtx).Table(updatedSchema.ReportConfigurationsTableName).Select("serialized")
+		rows, err := query.Rows()
+		if err != nil {
+			return errors.Wrapf(err, "failed to iterate table %s", "report_configurations")
+		}
+		defer func() { _ = rows.Close() }()
+		var convertedReportConfigs []*updatedSchema.ReportConfigurations
+		var reportSnapshots []*updatedSchema.ReportSnapshots
+		var reportNotifiers []*updatedSchema.ReportConfigurationsNotifiers
+		var errList errorhelpers.ErrorList
+		for rows.Next() {
+			// convert to report config proto
+			var reportConfig *updatedSchema.ReportConfigurations
+			if err = query.ScanRows(rows, &reportConfig); err != nil {
+				return errors.Wrapf(err, "failed to scan rows")
+			}
+			reportConfigProto, err := updatedSchema.ConvertReportConfigurationToProto(reportConfig)
+			if err != nil {
+				return errors.Wrapf(err, "failed to convert report config  %+v to proto", reportConfigProto)
+			}
+			// skip if version=0 and scope id is not nil since it is v2 config created in tech preview
+			if reportConfigProto.Version == 2 || (reportConfigProto.Version == 0 && reportConfigProto.GetResourceScope() != nil) {
```

Comment:
If version is already == 2, no need to do anything with that config. Remove `reportConfigProto.Version == 2` and before this if condition, add
```
if reportConfigProto.Version == 2 {
    continue
}
```